src/MongoDB.Bson/ObjectModel/BsonBinaryData.cs

```diff
@@ -70,25 +70,39 @@ public BsonBinaryData(byte[] bytes, BsonBinarySubType subType, GuidRepresentatio
                     var message = string.Format(
                         "Length must be 16, not {0}, when subType is {1}.",
                         bytes.Length, subType);
-                    throw new ArgumentException(message);
+                    throw new ArgumentException(message, nameof(bytes));
                 }
-                var expectedSubType = (guidRepresentation == GuidRepresentation.Standard) ? BsonBinarySubType.UuidStandard : BsonBinarySubType.UuidLegacy;
```

Comment:
This simple ternary expression fails to handle invalid guidRepresentation so it has been converted to a switch statement.